Only try to call specialized findin method for sorted input when elements are Real.

andreasnoack opened the pull request:
This should handle the issue mentioned in https://github.com/JuliaLang/julia/pull/21934#issuecomment-302873055. Eventually, we might want to handle this in a smarter way. See https://github.com/JuliaLang/julia/pull/21999.

andreasnoack committed:
Only try to call specialized findin method for sorted input when
elements are Real.

ararslan commented on base/array.jl:
````diff
@@ -1791,13 +1791,16 @@ julia> findin(a,b) # 10 is the only common element
 ```
 """
-function findin(a, b)
+function findin(a::Array{<:Real}, b::Union{Array{<:Real,Real}})
````
I don't understand the union here. Did you mean `Union{Array{<:Real}, Real}`? Otherwise this is `Array{T, Real} where T<:Real`.

andreasnoack replied:
You are right. It is a typo.

andreasnoack committed:
Fix termination cirteria in _sortedfindin

Use total order `isless` for tests

Add tests

andreasnoack committed:
Define eachindex for numbers

andreasnoack commented:
I've updated this PR with fixes to the algorithm for the issue pointed out in https://github.com/JuliaLang/julia/pull/21934#issuecomment-303033533 and added tests. I had to define `eachindex(Number)` to make this work but since `Number`s work with `getindex`, I believe this is the right thing to do anyway.

andreasnoack committed:
Add a few comments to the algorithm

tkelman commented on base/array.jl:
```diff
@@ -1734,6 +1734,9 @@ function _findin(a, b)
 end
 
+# If two collections are already sorted, findin can be computed with
+# a single traversal of the two collections. This is much fast than
```
faster than
